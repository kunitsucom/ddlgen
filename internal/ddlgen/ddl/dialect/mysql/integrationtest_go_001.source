package main

type (
	// User is a user.
	//
	// myddl:      table: `users`
	// myddl: constraint: UNIQUE(`name`)
	// myddl:      index: CREATE INDEX `index_users_by_name` ON `users` (`name`)
	User struct {
		// UserID is a user ID.
		UserID string `dbtest:"user_id" myddl:"VARCHAR(36)  NOT NULL" pkey:"true"`
		// Name is a user name.
		Name string   `dbtest:"name"    myddl:"VARCHAR(255) NOT NULL"`
		// Email is a user email.
		Email string  `dbtest:"email"   myddl:"VARCHAR(255) NOT NULL"`
		// Age is a user age.
		Age int       `dbtest:"age"     myddl:"INT          NOT NULL"`
		// Ignore is a ignore field.
		Ignore string `dbtest:"-"       myddl:"-"`
	}

	// Users is a user array.
	// This type is expected not to be detected.
	//
	// myddl: table: `UserArrays`
	Users []*User

	// UserGroup is a user group.
	//
	// myddl:table:
	UserGroup struct {
		// ID is a group ID.
		ID string   `dbtest:"id"   myddl:"VARCHAR(36)  NOT NULL"`
		// Name is a group name.
		Name string `dbtest:"name" myddl:"VARCHAR(255) NOT NULL"`
	}

	// Author is a author.
	// myddl:
	Author struct {
		// ID is a author ID.
		AuthorID string `dbtest:"id"   myddl:"VARCHAR(36)  NOT NULL"`
		// Name is a author name.
		Name string     `dbtest:"name" myddl:"VARCHAR(255) NOT NULL"`
	}

	// Book is a book.
	//
	// myddl:table:`books`
	Book struct {
		// AuthorID is a book author.
		AuthorID string `myddl:"VARCHAR(255) NOT NULL" pkey:"true"`
		// ID is a book ID.
		ID string       `myddl:"VARCHAR(36)  NOT NULL" pkey:"true"`
		// Title is a book title.
		Title string    `myddl:"VARCHAR(255) NOT NULL"`
	}

	// Store
	// myddl: table: CREATE TABLE `stores`
	Store struct {
		// ID is a store ID.
		ID string   `dbtest:"id" pkey:"false"`
		// Name is a store name.
		Name string `dbtest:"name"`
	}
)

// myddl: index: `index_books_by_title` ON `books` (`title`)
