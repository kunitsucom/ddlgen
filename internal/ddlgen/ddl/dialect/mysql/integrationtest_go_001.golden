-- Code generated by ddlgen. DO NOT EDIT.
--

-- source: mysql/integrationtest_go_001.source:6
-- User is a user.
--
-- myddl:      table: `users`
-- myddl: constraint: UNIQUE(`name`)
-- NOTE: the "User" struct's "Ignore" field has a tag for column name (`dbtest:"-"`), so the field is ignored.
CREATE TABLE `users` (
    -- UserID is a user ID.
    `user_id` VARCHAR(36)  NOT NULL,
    -- Name is a user name.
    `name`    VARCHAR(255) NOT NULL,
    -- Email is a user email.
    `email`   VARCHAR(255) NOT NULL,
    -- Age is a user age.
    `age`     INT          NOT NULL,
    PRIMARY KEY (`user_id`),
    UNIQUE(`name`)
);

-- source: mysql/integrationtest_go_001.source:8
-- myddl:      index: CREATE INDEX `index_users_by_name` ON `users` (`name`)
CREATE INDEX `index_users_by_name` ON `users` (`name`);

-- source: mysql/integrationtest_go_001.source:25
-- Users is a user array.
-- This type is expected not to be detected.
--
-- myddl: table: `UserArrays`
-- ERROR: the comment (mysql/integrationtest_go_001.source:22) does not have struct fields for column type and constraint (`myddl:"<TYPE> [CONSTRAINT]"`), or the comment is not associated with struct.
CREATE TABLE `UserArrays` (
);

-- source: mysql/integrationtest_go_001.source:30
-- UserGroup is a user group.
--
-- myddl:table:
-- WARN: the comment (mysql/integrationtest_go_001.source:28) does not have a key for table (myddl: table: CREATE TABLE <table>), so the struct name "UserGroup" is used as the table name.
CREATE TABLE UserGroup (
    -- ID is a group ID.
    `id`   VARCHAR(36)  NOT NULL,
    -- Name is a group name.
    `name` VARCHAR(255) NOT NULL
);

-- source: mysql/integrationtest_go_001.source:39
-- Author is a author.
-- myddl:
-- WARN: the comment (mysql/integrationtest_go_001.source:38) does not have a key for table (myddl: table: CREATE TABLE <table>), so the struct name "Author" is used as the table name.
CREATE TABLE Author (
    -- ID is a author ID.
    `id`   VARCHAR(36)  NOT NULL,
    -- Name is a author name.
    `name` VARCHAR(255) NOT NULL
);

-- source: mysql/integrationtest_go_001.source:49
-- Book is a book.
--
-- myddl:table:`books`
CREATE TABLE `books` (
    -- WARN: the "Book" struct's "AuthorID" field does not have a tag for column name (`dbtest:"<ColumnName>"`), so the field name "AuthorID" is used as the column name.
    -- AuthorID is a book author.
    `AuthorID` VARCHAR(255) NOT NULL,
    -- WARN: the "Book" struct's "ID" field does not have a tag for column name (`dbtest:"<ColumnName>"`), so the field name "ID" is used as the column name.
    -- ID is a book ID.
    `ID`       VARCHAR(36)  NOT NULL,
    -- WARN: the "Book" struct's "Title" field does not have a tag for column name (`dbtest:"<ColumnName>"`), so the field name "Title" is used as the column name.
    -- Title is a book title.
    `Title`    VARCHAR(255) NOT NULL,
    PRIMARY KEY (`AuthorID`, `ID`)
);

-- source: mysql/integrationtest_go_001.source:60
-- Store
-- myddl: table: CREATE TABLE `stores`
CREATE TABLE `stores` (
    -- ERROR: the "Store" struct's "ID" field does not have a tag for column type and constraint (`myddl:"<TYPE> [CONSTRAINT]"`)
    -- WARN: the field "ID" does not have valid primary key tag (`pkey:"true"`), so the column is not used as primary key.
    -- ID is a store ID.
    `id`   DDLGEN_ERROR_STRUCT_FIELD_TAG_NOT_FOUND,
    -- ERROR: the "Store" struct's "Name" field does not have a tag for column type and constraint (`myddl:"<TYPE> [CONSTRAINT]"`)
    -- Name is a store name.
    `name` DDLGEN_ERROR_STRUCT_FIELD_TAG_NOT_FOUND
);

-- source: mysql/integrationtest_go_001.source:69
-- myddl: index: `index_books_by_title` ON `books` (`title`)
CREATE INDEX `index_books_by_title` ON `books` (`title`);

-- source: mysql/integrationtest_go_001.source:69
-- WARN: the comment (mysql/integrationtest_go_001.source:69) does not have a key for table (myddl: table: CREATE TABLE <table>), or the comment is not associated with struct.
